### DISCUSSION TITLE
hashFiles('file') fails for an existing file

### DISCUSSION THREAD

**Discussion by stomar** [community/community#25559] [Actions]
<p>Hi,</p>
<p>this is my first post, I hope it’s ok.</p>
<p>For caching I need a hash of my Gemfile.lock, which is located in the root dir of the repository. I tried:</p>
<pre><code>${{ hashFiles('Gemfile.lock') }}
</code></pre>
<p>which does not work, though the file <em>does</em> exist in the directory. The reported error is:</p>
<pre><code>##[error]The template is not valid. 'hashFiles(Gemfile.lock)' failed. Search pattern 'Gemfile.lock' doesn't match any file under '/home/runner/work/www.ruby-lang.org/www.ruby-lang.org'
</code></pre>
<p>It works when I use this pattern:</p>
<pre><code>${{ hashFiles('**/Gemfile.lock') }}
</code></pre>
<p>In my case, there is only one Gemfile.lock in the repository, so it’s not really a problem. But strictly speaking it’s not what I wanted.</p>
<p>Additionally, the returned hash is not the SHA-256 sum I get with “sha256sum Gemfile.lock” on my local (Ubuntu) machine, is that to be expected?</p>
<p>The log of the workflow run can found here:</p>
<p><a href="https://github.com/ruby/www.ruby-lang.org/commit/0cb75fafb86b1663f78e876282f5815401170795/checks?check_suite_id=303279531" title="workflow log" rel="nofollow noopener">https://github.com/ruby/www.ruby-lang.org/commit/0cb75fafb86b1663f78e876282f5815401170795/checks?check_suite_id=303279531</a></p>
<p>Thanks,</p>
<p>Marcus (<a class="mention" href="/stomar">@stomar</a>)</p>

**Comment by imhoffd** [comment]
<p>Getting this issue for macOS. Additionally, using **/ with hashFiles also seems to fail outright on Windows:</p>
<pre><code>Search pattern '**/package-lock.json' doesn't match any file under 'd:\a\project\project'
</code></pre>
<p>I’m guessing the glob library isn’t normalizing directory separators.</p>

**Comment by thboop** [comment, marked as answer]
<p>We are aware of both of these issues and fixes are being worked on</p>
<p><a href="https://github.com/actions/cache/issues/39#issuecomment-552252932" rel="nofollow noopener">https://github.com/actions/cache/issues/39#issuecomment-552252932</a></p>

**Comment by ericsciple** [comment]
<p>the fix should be rolled out everywhere now</p>

**Comment by stomar** [comment]
<p>Yes, it works now. Thank you!</p>
